projects/backend/src/main/java/backend/security/JwtAuthorizationFilter.java

```diff
@@ -0,0 +1,59 @@
+package backend.security;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import javax.servlet.FilterChain;
+
+import javax.servlet.ServletException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+public class JwtAuthorizationFilter extends BasicAuthenticationFilter {
+  public JwtAuthorizationFilter(AuthenticationManager authManager) {
+    super(authManager);
+  }
+
+  @Override
+  protected void doFilterInternal(HttpServletRequest req,
+                                  HttpServletResponse res,
+                                  FilterChain chain) throws IOException, ServletException {
+    String header = req.getHeader(SecurityConstants.HEADER_STRING);
+
+    if (header == null || !header.startsWith(SecurityConstants.TOKEN_PREFIX)) {
+      chain.doFilter(req, res);
+      return;
+    }
+
+    UsernamePasswordAuthenticationToken authentication = getAuthentication(req);
+
+    SecurityContextHolder.getContext().setAuthentication(authentication);
+    chain.doFilter(req, res);
+  }
+
+  private UsernamePasswordAuthenticationToken getAuthentication(HttpServletRequest request) {
+    String token = request.getHeader(SecurityConstants.HEADER_STRING);
+    if (token != null) {
+      // parse the token.
+      String user = JWT.require(Algorithm.HMAC512(SecurityConstants.SECRET.getBytes()))
+          .build()
+          .verify(token.replace(SecurityConstants.TOKEN_PREFIX, ""))
+          .getSubject();
+
+      if (user != null) {
+        return new UsernamePasswordAuthenticationToken(user, null, new ArrayList<>());
+      }
+      return null;
+    }
+    return null;
+  }
```

Comment:
If it's possible do not return ```null```, prefare usage of a [Optional]( https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html).